segbot_arm_joystick/src/segbot_arm_joystick.cpp

```diff
@@ -0,0 +1,505 @@
+#include <ros/ros.h>
+#include <ros/package.h>
+
+//used for registering Ctrl-C events
+#include <signal.h>
+
+//ROS messages
+#include <sensor_msgs/JointState.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TwistStamped.h>
+
+//ROS action client used for calling actions
+#include <actionlib/client/simple_action_client.h>
+
+//JACO messages and actions
+#include <jaco_msgs/FingerPosition.h>
+#include <jaco_msgs/HomeArm.h>
+
+//our own arm library 
+#include <segbot_arm_manipulation/arm_utils.h>
+
+#include <sensor_msgs/Joy.h>
+#include <bwi_services/SpeakMessage.h>
```

Comment:
There appear to be missing dependencies on ``segbot_services``, in both ``CMakeLists.txt`` and ``package.xml``.

See: http://docs.ros.org/indigo/api/catkin/html/howto/format1/cpp_msg_dependencies.html